src/components/inspectors/TimerExpression.vue

```diff
@@ -0,0 +1,381 @@
+    selectedWeekdays() {
+      const selected = [];
+      this.weekdays.forEach(weekday => {
+        weekday.selected ? selected.push(weekday.day) : null;
+      });
+      return selected;
+    },
+    sameDay() {
+      return this.selectedWeekdays.length === 1 && this.weekdays[this.startDate.getDay()].selected;
+    },
+  },
+  watch: {
+    value(value) {
+      this.parseTimerConfig(value);
+    },
+  },
+  methods: {
+    currentDaySelected(day) {
+      return this.startDate.getDay() === (day.day % 7);
+    },
+    updateStartDate(date) {
+      this.startDate = date;
+      this.update();
+    },
+    updateEndDate(date) {
+      this.endDate = date;
+      this.update();
+    },
+    update() {
+      this.$emit('input', this.expression);
+    },
+    parseDateExpression(exp) {
+      const date = new Date(exp);
+      if (isNaN(date.getTime())) {
+        throw 'Invalid Date';
+      }
+      return moment(date);
+    },
+    parseTimerConfig(value) {
+      this.resetTimerExpression();
+      if (!value) {
+        return;
+      }
+      try {
+        let date, hasStartDate = false;
+        const expression = value.split('|');
+        expression.forEach(exp => {
+          if (exp.substr(0, 1) !== 'R') {
+            date = this.parseDateExpression(exp);
+            this.startDate = date.toDate();
+            this.startTime = date.format('HH:mm');
+          } else {
+            let match = exp.match(/R(\d*)\/([^/]+)\/P(\d+)(\w)(?:\/([^/]+))?/);
+            if (match) {
+              this.times = match[1] || '1';
+              date = this.parseDateExpression(match[2]);
+              hasStartDate ? null : this.startDate = date.toDate();
+              hasStartDate ? null : this.startTime = date.format('HH:mm');
+              this.repeat = match[3];
+              this.periodicity = Object.keys(periods).find(key => periods[key] === match[4]);
+              this.endDate = match[5] ? this.parseDateExpression(match[5]).toDate() : new Date();
+              this.ends = !match[5] ? !match[1] ? 'never' : 'after' : 'ondate';
+              if (this.periodicity === 'week') {
+                // Note this.weekday array must start with Sunday
+                this.weekdays[date.get('day')].selected = true;
+              }
+            }
+          }
+          hasStartDate = true;
+        });
+      } catch (invalidExpression) {
+        this.resetTimerExpression();
+      }
+    },
+    resetTimerExpression() {
+      this.startDate = new Date();
+      this.startTime = '00:00';
+      this.times = '1';
+      this.repeat = '1';
+      this.periodicity = 'week';
+      this.endDate = new Date();
+      this.ends = 'never';
+      this.weekdays.forEach(weekday => weekday.selected = false);
+    },
+    clickWeekDay(weekday) {
+      weekday.selected = !weekday.selected;
+    },
+    makeTimerConfig() {
+      const expression = [];
+      if (this.periodicity === 'week' && this.selectedWeekdays.length > 0 && !this.sameDay) {
```

Comment:
For readability, could you extract this condition into a function ?

Reply:
Moved to **hasMultipleWeekdaySelected**